File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) 
    {
        // let's use two pointer starting from the maximum length
        int r = height.length-1;
        int l = 0;
        int maxWater = 0;
        
        while(l<r)
        {
            if(height[r] > height[l])
            {
                maxWater = Math.max(maxWater,(r-l)*height[l]);
                l++;
            }
            else
            {
                maxWater = Math.max(maxWater,(r-l)*height[r]);
                r--;
            }
        }
        return maxWater;
    }
}

File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        l,r=0,len(height)-1
        maxArea = 0;
        while l < r:
            maxArea = max(maxArea,(r-l)*min(height[l],height[r]))
            if height[l] > height[r]: 
                r-=1;
            else:
                l+=1
        return maxArea;
                
            
        